dbt/clients/jinja.py

```diff
@@ -37,13 +41,28 @@ def _parse(self, source, name, filename):
             jinja2._compat.encode_filename(filename)
         ).parse()
 
+    def _compile(self, source, filename):
+        """Override jinja's compilation to stash the rendered source inside
+        the python linecache for debugging.
+        """
+        if filename == '<template>':
+            # make a better filename
+            filename = 'dbt-{}'.format(
+                codecs.encode(os.urandom(12), 'hex').decode('ascii')
+            )
+            # encode, though I don't think this matters
+            filename = jinja2._compat.encode_filename(filename)
+            # put ourselves in the cache using the 'lazycache' method
+            linecache.cache[filename] = (lambda: source,)
+
+        return super(MacroFuzzEnvironment, self)._compile(source, filename)
```

Comment:
i'd like to talk to you about this for 5min. i understand the idea on a basic level but am not familiar with the python internals here